### DISCUSSION TITLE
Is there a way to execute the Git grep command for code search on GitHub server?

### DISCUSSION THREAD

**Discussion by rahulsb** [community/community#22745] [New to GitHub]
<p>Hi,</p>
<p>I know that there are GitHub code search REST APIs available however for my requirement those are not fitting in.</p>
<p>I would like to search code in non default branches and tags as well which is not supported by GitHub REST API.</p>
<p>The Git grep command fits well for my requirement ,<br>
example:<br>
git grep -n3 https 1.0 *.js<br>
git grep -n3 https master-1 *.java</p>
<p>Questions:<br>
I am not able to figure out how to fire this git command directly on the server.<br>
Obviously I could fire this in my Git CMD client.<br>
Is there a way I could directly fire this command on GitHub server .</p>
<p>Thanks in Advance.<br>
Rahul</p>

**Comment by yamiacat** [comment, marked as answer]
<p>Hi <a class="mention" href="/rahulsb">@rahulsb</a>! :wave: Welcome to the Community!</p>
<p>I’m afraid this is not possible, sorry! You’d need to clone a local copy to be able to grep all branches.</p>
